server/api/observation/index.js

```diff
@@ -3,60 +3,72 @@ var db = require('../../db');
 
 // Return a collection of observations selected corresponding to an eventId
 router.get('/:eventId', function(req, res) {
-  db.Observation.findAll({
-    where: {
-      EventId: req.params.eventId
-    },
-    include: {
-      model: db.User
-    },
-    order: 'id DESC'
-  }).then(function(event) {
-    res.json(event);
-  });
+  if (!isNaN(req.params.eventId)) {
+    // eventId is a number
+    db.Observation.findAll({
+      where: {
+        EventId: +req.params.eventId
+      },
+      include: {
+        model: db.User
+      },
+      order: 'id DESC'
+    }).then(function(observations) {
+      res.json(observations);
+    });
+  } else {
+    // eventId is not a number
+    res.sendStatus(400);
+  }
 });
 
 // Create a new observation and return it
 router.post('/create', function(req, res) {
- // Create and add the observation to the db
- db.Observation.create({
-   content: req.body.content || '',
-   completed: req.body.completed || false,
-   rawImage: req.body.rawImage || undefined,
-   UserId: req.body.userId,
-   EventId: req.body.eventId
- }).then(function(observation){
-    //Find the event to which the observation belongs to
-    db.Event.findOne({
+  // Check if userId and eventId exist and are numbers
+  if (req.body.userId && !isNaN(req.body.userId) && req.body.eventId && !isNaN(req.body.eventId)) {

```

Comment:
I think checking for !isNaN(req.body.userId) && !isNaN(req.body.userId) might be enough